ddl/ddl_api.go

```diff
@@ -2705,6 +2705,15 @@ func (d *ddl) ModifyColumn(ctx sessionctx.Context, ident ast.Ident, spec *ast.Al
 			if err := checkAutoIncrementRef(specNewColumn.Name.Name.L, dependColNames, t.Meta()); err != nil {
 				return errors.Trace(err)
 			}
+
+			// If there is an index on the generated column which we are trying to modify, return an error.
```

Comment:
Wrap this `if option.Tp == ast.ColumnOptionGenerated` code block into a function like `changeGeneratedColumns`